src/core/environment.go

```diff
@@ -112,6 +113,36 @@ func (env *EnvironmentType) NewHostInfo(agentVersion string, tags *[]string, con
 	return env.host
 }
 
+// getUnixName returns details about this operating system formatted as "sysname
+// nodename release version machine". Returns "" if unix name cannot be
+// determined.
+//
+//   - sysname: Name of the operating system implementation.
+//   - nodename: Network name of this machine.
+//   - release: Release level of the operating system.
+//   - version: Version level of the operating system.
+//   - machine: Machine hardware platform.
+//
+// Different platforms have different [Utsname] struct definitions.
+//
+// TODO :- Make this function platform agnostic to pull uname (uname -a).
+//
+// [Utsname]: https://cs.opensource.google/search?q=utsname&ss=go%2Fx%2Fsys&start=1
+func getUnixName () string {
+	var utsname unix.Utsname
+	err := unix.Uname(&utsname)
+	if(err != nil) {
+		log.Warnf("Unable to read Uname. Error: %v", err)
+		return ""
+	}
+	sysName := string(utsname.Sysname[:bytes.IndexByte(utsname.Sysname[:], 0)])
```

Comment:
corner case: are these `utsname` fields guaranteed to have a 0 byte? If not, `IndexByte` can return -1, and `slice[:-1]` throws a runtime error.

if we think that's possible, maybe a helper function?

```go
toStr := func(buf []byte) string {
  idx := bytes.IndexByte(buf, 0)
  if idx == -1 {
    return "unknown"
  }
  return string(bytes[:idx))
}

sysName = toStr(utsname.Sysname[:])
// ... etc
```

Might be too much of a corner case, LGTM otherwise.

Reply:
Have addressed this corner case.